code/game/machinery/teleporter.dm

```diff
@@ -169,6 +184,12 @@
 
 	. = L
 
+/obj/machinery/computer/teleporter/proc/change_freq(var/mob/user)
+	var/newfreq = input("Input a new frequency for the teleporter", "Frequency", null) as num
```

Comment:
you didn't give it `as null|num` so you won't get a cancel button, and the null check will never do anything
you also need to check whether the user is still able to interact with the teleporter after the input, because it `sleep()`s

Reply:
That bit was mostly copied over from beacons, so I guess that explains why they never had a cancel prompt all this time either. To be honest I'm not actually sure _how_ to check if the user's still able to interact with the teleporter after the input.

Reply:
Look at what machinery/Topic does, that's what's needed for someone to be able to interact with a machine. How to call that from there is left as an exercise to the reader

Reply:
Fixed (albeit not as elegantly as I'd like), along with similar pre-existing bugs with radio beacons.